source/wp-content/themes/wporg-parent-2021/sass/base/_elements.scss

```diff
@@ -125,6 +125,16 @@ input[type="checkbox"] + label {
 	line-height: 1em; // stylelint-disable-line declaration-property-unit-allowed-list
 }
 
+// Headings.
+// Add margin to default h2 & h3 (no font size changes).
+.wp-site-blocks h2:not([class*="-font-size"], [style*="font-size"]) {
```

Comment:
I can't remember off the top of my head, but do we control the class `.wp-site-blocks`? It would be nice to know that this can't be removed or renamed by gutenberg and break our styling.

Reply:
We don't control `.wp-site-blocks`, but we're already using it for some other basic layout styles, so if it disappears we'll know pretty quickly. This class is also being used in at least Twenty Twenty-Two, so we're not going totally against the grain. The issue we ran into with the other container class was that it was a generated class, so GB wasn't expecting anyone to use it.

> I'm also a bit wary of using Gutenberg styles in the :not clause although I can't think of a better solution. 🤔

I think we can be fairly confident these aren't going to change, though. I'm pretty sure it's been `has-*-font-size` since that feature started, and `style="font-size…"` when it's a custom size seems unlikely to change.